src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -0,0 +1,229 @@
+                                            kubernetesAPIObject.setType(dsw.getDataSourceWorkloadType());
+                                            kubernetesAPIObject.setNamespace(namespace.getDataSourceNamespaceName());
+                                            kubernetesAPIObjectList.add(kubernetesAPIObject);
+                                            createExperimentAPIObject.setKubernetesObjects(kubernetesAPIObjectList);
+                                            RecommendationSettings rs = new RecommendationSettings();
+                                            rs.setThreshold(0.1);
+                                            createExperimentAPIObject.setRecommendationSettings(rs);
+                                            TrialSettings trialSettings = new TrialSettings();
+                                            trialSettings.setMeasurement_durationMinutes("15min");
+                                            createExperimentAPIObject.setTrialSettings(trialSettings);
+                                            List<KruizeObject> kruizeExpList = new ArrayList<>();
+
+                                            createExperimentAPIObject.setExperiment_id(Utils.generateID(createExperimentAPIObject.toString()));
+                                            createExperimentAPIObject.setStatus(AnalyzerConstants.ExperimentStatus.IN_PROGRESS);
+
+                                            try {
+                                                ValidationOutputData output = new ExperimentDBService().addExperimentToDB(createExperimentAPIObject);
+                                                if (output.isSuccess()) {
+                                                    jobStatusMap.get(jobID).getData().getExperiments().setNewExperiments(
+                                                            appendExperiments(jobStatusMap.get(jobID).getData().getExperiments().getNewExperiments(), experiment_name)
+                                                    );
+                                                }
+                                                recommendationsRequiredExperiments.add(experiment_name);
+                                            } catch (Exception e) {
+                                                LOGGER.info(e.getMessage());
+                                            }
+                                        }
+                                    }
+
+                                }
+                            }
+                        }
+                    }
+                }
+                jobStatusMap.get(jobID).setStatus("INPROGRESS");
+                jobStatusMap.get(jobID).getData().getRecommendations().getData().setInqueue(recommendationsRequiredExperiments);
+                jobStatusMap.get(jobID).getData().getRecommendations().setTotalCount(recommendationsRequiredExperiments.size());
+
+            }
+            ExecutorService executor = Executors.newFixedThreadPool(3);
```

Comment:
Why hardcoding to 3 threads? How will this affect scalability

Reply:
Now I made this configurable vai env .  Adjust the thread pool size based on the kruize pod count 
at most three threads will be allowed to execute concurrently.

in future we can use inline with Keda plugin and have  dynamic number of threads to scale based on demand.